tests/suites/test_suite_platform.function

```diff
@@ -32,53 +32,58 @@ void sleep_ms(int milliseconds)
     usleep(milliseconds * 1000);
 #endif
 }
-#endif
-
-/* END_HEADER */
-
-/* BEGIN_DEPENDENCIES */
 
-/* END_DEPENDENCIES */
 
-/* BEGIN_CASE depends_on:MBEDTLS_HAVE_TIME */
-void time_get_milliseconds()
+int test_time_delay_milliseconds_once(int delay_ms)
 {
     mbedtls_ms_time_t current = mbedtls_ms_time();
-    (void) current;
-    /* This goto is added to avoid warnings from the generated code. */
-    goto exit;
+    mbedtls_ms_time_t end;
+    mbedtls_ms_time_t elapsed_ms;
+
+    sleep_ms(delay_ms);
+
+    end = mbedtls_ms_time();
+    elapsed_ms = end - current;
+
+    return elapsed_ms > 0 && elapsed_ms >= delay_ms && \
+           elapsed_ms < 4000 + delay_ms;
```

Comment:
Please use separate conditions and arrange that if the test fails, the failure message indicates which condition failed. Ideally, display the numerical values.

With a single iteration of the test, that can be written as follows:
```
TEST_LE_U(1, elapsed_ms);
TEST_LE_U(delay_ms, elapsed_ms);
TEST_LE_U(elapsed_ms, 4000 + delay_ms - 1);
```

Note that when a function is used in a loop, you can use `mbedtls_test_set_step` to show the iteration number in the failure message. However, that works nicely when we want all iterations to succeed, not when we want at least one iteration to succeed.

I can think of a few solutions that involve either making the code here a lot more complicated, or adding a weird feature to the interface of test failure that allows resetting a test to pass. I don't really like either of those.

In any case, I don't like repeating the test on failure. If failure is rare but can happen, then failure multiple times can happen. They are not uncorrelated failures: they typically happen only in certain environments or under a certain load pattern, but when the conditions are met, the chance of failure at each iteration is high. So I think we should only run one iteration, and make it loose enough so that it will always pass even on a highly loaded CI.

Reply:
>In any case, I don't like repeating the test on failure. If failure is rare but can happen, then failure multiple times can happen. They are not uncorrelated failures: they typically happen only in certain environments or under a certain load pattern, but when the conditions are met, the chance of failure at each iteration is high. So I think we should only run one iteration, and make it loose enough so that it will always pass even on a highly loaded CI.

This PR try to ignore three cases:  `signal interrupt`, `discontinues jumps` and `incremental adjustment`. I think the conditions won't always be met. If the three conditions are always met, the platform might have some fatal errors.

For other cases, the conditions are always met, the test will raise failure. It is expected.

1.  signal interrupt: That is introduced by `sleep_ms()` . To resolve that, we can call `mbedtls_time` or `mbedtls_time_ms` again. It looks strange. When first iteration fails, the signal has been consumed, next iteration will pass.
1. `discontinues jumps`:  Linux `time()`'s clock source is `CLOCK_REALTIME`, it can be manually changed. When it happens in first iteration, second iteration will be success. This case won't happen in CI environment, but it might happen in local environment.
1. `incremental adjustment`: when incremental adjustments are performed, the iteration will be failed. And I think it won't be performed or cause test failure within a short time. 
   1. If it is start and end within first iteration, second iteration will be success . 
   1. If it is start in first iteration and end in second iteration, third iteration will be success 

So, 3 iterations are introduced in this PR. 
I ignore 2 cases. The conditions happen together and the time of `incremental adjustment` is longer than 2 iteration. Both cases are very rare. 


With one iteration solution, we have to adjust tolerance window.  It can not cover all conditions
1. signal interrupt: the negative tolerance MUST be same with delay time. It is not reasonable.
2.  `discontinues jumps`: it can not be covered. We can not predicate the new time.
3.  `incremental adjustment`: Mostly it is caused by NTP, we should set the tolerance window to ±8.6 seconds( suppose NTP sync up time once per day). [Typical crystal RTC accuracy specifications are from ±100 to ±20 parts per million (8.6 to 1.7 seconds per day)](https://en.wikipedia.org/wiki/Real-time_clock)